apps/api/src/app/notification-template/usecases/create-notification-template/create-notification-template.usecase.ts

```diff
@@ -19,12 +19,19 @@ export class CreateNotificationTemplate {
     const contentService = new ContentService();
     const variables = contentService.extractMessageVariables(command.messages);
 
+    const templateCheck = await this.notificationTemplateRepository.findByTriggerIdentifier(
+      command.organizationId,
+      `${slugify(command.name, {
+        lower: true,
+        strict: true,
+      })}`
+    );
     const trigger: INotificationTrigger = {
       type: TriggerTypeEnum.EVENT,
       identifier: `${slugify(command.name, {
         lower: true,
         strict: true,
-      })}-${shortid.generate()}`,
+      })}${!templateCheck ? '' : '-' + shortid.generate()}`,
```

Comment:
Yeah, this is exactly what we needed! only add the shortid in case we have collision, other wise just use the plain slug.